### ISSUE TITLE
Bullets stay in the air after save/quit and loading

### ISSUE THREAD

**Issue by Alicecomma** [NoImageAvailable/CombatExtended#513]
My initial guess was the bullets aren't exposed or spawned properly on game load, such that they are not "Launched" but just appear until something launches them (which never happens).

However, it appears errors during TakeDamage cause the bullet to stop its Impact() and thus never reach Destroy().

Representative error (error fixed in #749):
```
Exception ticking Bullet_762x54mmR_HP16135 (at (85, 0, 188)): System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: index
at System.Collections.Generic.List`1<RimWorld.Apparel>.get_Item (int) <0x00083>
at CombatExtended.ArmorUtilityCE.GetAfterArmorDamage (Verse.DamageInfo,Verse.Pawn,Verse.BodyPartRecord,bool&,bool&,bool&) <0x00b82>
at CombatExtended.Harmony.Harmony_DamageWorker_AddInjury_ApplyDamageToPart.ArmorReroute (Verse.Pawn,Verse.DamageInfo&,bool&,bool&) <0x000c9>
at (wrapper dynamic-method) Verse.DamageWorker_AddInjury.ApplyDamageToPart_Patch1 (object,Verse.DamageInfo,Verse.Pawn,Verse.DamageWorker/DamageResult) <0x00127>
at Verse.DamageWorker_AddInjury.ApplyToPawn (Verse.DamageInfo,Verse.Pawn) <0x0033e>
at Verse.DamageWorker_AddInjury.Apply (Verse.DamageInfo,Verse.Thing) <0x000be>
at Verse.Thing.TakeDamage (Verse.DamageInfo) <0x002dd>
at CombatExtended.BulletCE.Impact (Verse.Thing) <0x0050d>
at CombatExtended.ProjectileCE.TryCollideWith (Verse.Thing) <0x008c2>
at CombatExtended.ProjectileCE.CheckCellForCollision (Verse.IntVec3) <0x0090a>
at CombatExtended.ProjectileCE.CheckForCollisionBetween () <0x003af>
at CombatExtended.ProjectileCE.Tick () <0x001cd>
at Verse.TickList.Tick () <0x00487>

Verse.Log:Error(String, Boolean)
Verse.TickList:Tick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()
```

Error with a more modded setup (iceball3, Discord 3864):
(NoImageAvailable: My guess is muscle spasms and bullet damage trying to down at once)
```
Exception ticking Bullet_12Gauge_Beanbag59705: System.NullReferenceException: Object reference not set to an instance of an object
at Verse.Pawn_HealthTracker.MakeDowned (System.Nullable`1<Verse.DamageInfo>,Verse.Hediff) <0x00780>
at (wrapper dynamic-method) Verse.Pawn_HealthTracker.CheckForStateChange_Patch0 (object,System.Nullable`1<Verse.DamageInfo>,Verse.Hediff) <0x000de>
at Verse.Pawn_HealthTracker.AddHediff (Verse.Hediff,Verse.BodyPartRecord,System.Nullable`1<Verse.DamageInfo>) <0x0007c>
at Verse.Pawn_HealthTracker.PostApplyDamage (Verse.DamageInfo,single) <0x00236>
at Verse.Pawn.PostApplyDamage (Verse.DamageInfo,single) <0x000dd>
at Verse.Thing.TakeDamage (Verse.DamageInfo) <0x00407>
at CombatExtended.BulletCE.Impact (Verse.Thing) <0x003d4>
at CombatExtended.ProjectileCE.TryCollideWith (Verse.Thing) <0x00676>
at CombatExtended.ProjectileCE.CheckCellForCollision (Verse.IntVec3) <0x00690>
at CombatExtended.ProjectileCE.CheckForCollisionBetween () <0x00297>
at CombatExtended.ProjectileCE.Tick () <0x00153>
at Verse.TickList.Tick () <0x002c6>
```

Simplest possible fix is to try { } catch { Destroy() } and ignore any errors. However, this would obviously _ignore any errors_.

**Comment by Alicecomma** [comment]
The first representative error is actually fixed by https://github.com/NoImageAvailable/CombatExtended/pull/749/commits/aef917c0d88a6cdd8ed3d1af3ae75f45ecc6c4f6

The problem here was that GetAfterArmorDamage's loop through WornApparel would apply damage to individual members of that list, sometimes destroying the Thing and thus removing it from WornApparel. For some reason, "deflected" items are checked twice in the loop, meaning the loop would check the same index of WornApparel twice, erroring out if this was the last index in WornApparel.

The second error would not have been observed without this bug, but also the visual part of the bug itself (showing the bullet) is absolutely unnecessary for understanding what's going on.

I'd suggest try {} catch (Exception e) { Destroy{}; Log.Error(e); }.

**Comment by Alicecomma** [comment]
![image](https://user-images.githubusercontent.com/4662406/64032330-828fee80-cb4a-11e9-8df5-2b5c7f9af8a7.png)

**Comment by Alicecomma** [comment]
Merged in #769
